pkg/cluster/private_registry.go

```diff
@@ -34,15 +34,31 @@ func GetPrivateRepo(cluster *v3.Cluster) *rketypes.PrivateRegistry {
 	return nil
 }
 
+// TransformProvV2RegistryCredentialsToDockerConfigJSON transforms a ProvV2 registry secret into a credentialprovider.DockerConfigJSON secret
+func TransformProvV2RegistryCredentialsToDockerConfigJSON(url string, registryCredentials *corev1.Secret) (map[string][]byte, error) {
+	username := string(registryCredentials.Data["username"])
+	password := string(registryCredentials.Data["password"])
+	authConfig := credentialprovider.DockerConfigJSON{
+		Auths: credentialprovider.DockerConfig{
+			url: credentialprovider.DockerConfigEntry{
+				Username: username,
+				Password: password,
+			},
+		},
+	}
+	j, err := json.Marshal(authConfig)
+	return map[string][]byte{".dockerconfigjson": j}, err
+}
+
 func GenerateClusterPrivateRegistryDockerConfig(cluster *v3.Cluster) (string, error) {
 	if cluster == nil {
 		return "", nil
 	}
 	return GeneratePrivateRegistryDockerConfig(GetPrivateRepo(cluster), nil)
 }
 
-// This method generates base64 encoded credentials for the registry
-func GeneratePrivateRegistryDockerConfig(privateRegistry *rketypes.PrivateRegistry, registrySecret *corev1.Secret) (string, error) {
+// GeneratePrivateRegistryDockerConfig generates base64 encoded credentials for the provided registry
+func GeneratePrivateRegistryDockerConfig(privateRegistry *rketypes.PrivateRegistry, registrySecret map[string][]byte) (string, error) {
```

Comment:
If `".dockerconfigjson"` is the only key accessed, you could make the following change `registrySecret map[string][]byte -> dockerConfigJSON string`, this is really a nit though since the original ask was addressed.

Reply:
Addressed, though instead of a string i've used a byte slice as I would just have to convert the string back into a byte slice to unmarshal the json